bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/protocol/ascii/RotelAsciiV2ProtocolHandler.java

```diff
@@ -35,15 +36,22 @@ public class RotelAsciiV2ProtocolHandler extends RotelAbstractAsciiProtocolHandl
 
     private static final char CHAR_END_RESPONSE = '$';
 
+    private static final Set<String> KEYSET = Set.of(KEY_DISC_NAME, KEY_DISC_TYPE, KEY_TRACK_NAME, KEY_TIME, KEY_FM_RDS,
+            KEY_DAB_STATION);
+
     private final Logger logger = LoggerFactory.getLogger(RotelAsciiV2ProtocolHandler.class);
 
+    private boolean searchKey = true;
+    private boolean variableLength;
+    private boolean prevIsEndCharacter;
```

Comment:
Forgive me, but too tired right now to understand algorithm/parser below: Why are these instance variables?

Reply:
In case the reading thread handles the response in multiple "blocs".

Reply:
So the order is guaranteed and there is no need to "reset" when reaching the end of the full response?

Reply:
There is a reset done in the code (when calling `resetDataBuffer()`) to start with the handling of a new response.